app/models/data_source.rb

```diff
@@ -1,87 +1,29 @@
-require "active_record/connection_adapters/redshift_adapter"
-
 class DataSource < ApplicationRecord
-
-  validates :name, :adapter, :host, :dbname, :user, presence: true
+  validates :name, :adapter, presence: true
+  validates :host, :dbname, :user, presence: true, if: :standard_adapter?
 
   has_many :ignored_tables
 
   has_one :database_memo, class_name: "DatabaseMemo", foreign_key: :name, primary_key: :name, dependent: :destroy
+  has_one :bigquery_config, dependent: :destroy
+  accepts_nested_attributes_for :bigquery_config, update_only: true
```

Comment:
Sorry, I don't like `accepts_nested_attributes_for`.
I don't want to merge this change...

Reply:
ok. I'll remove it.